Naming/SmartUniqueNamer.php

```diff
@@ -17,10 +17,45 @@ public function name($object, PropertyMapping $mapping): string
     {
         $file = $mapping->getFile($object);
         $originalName = $file->getClientOriginalName();
-        $originalExtension = \pathinfo($originalName, PATHINFO_EXTENSION);
+        $originalExtension = \strtolower(\pathinfo($originalName, PATHINFO_EXTENSION));
         $originalBasename = \basename($originalName, '.'.$originalExtension);
         $originalBasename = Transliterator::transliterate($originalBasename);
+        $uniqId = \str_replace('.', '', \uniqid('-', true));
+        $uniqBasename = \sprintf('%s%s', $originalBasename, $uniqId);
+        $uniqExtension = \sprintf('%s.%s', $uniqId, $originalExtension);
+        $smartName = \sprintf('%s%s', $originalBasename, $uniqExtension);
 
-        return \sprintf('%s%s.%s', $originalBasename, \str_replace('.', '', \uniqid('-', true)), $originalExtension);
+        // Check if smartName is an acceptable size (some filesystems accept a max of 255)
+        if (\strlen($smartName) <= 255) {
+            return $smartName;
+        }
+
+        // Check if the unique extension will fit into 255
+        // 254 to account for a one letter basename
+        if (\strlen($uniqExtension) <= 254) {
+            // Resize the basename to fit into 255 alongside the unique ID and extension
+            $shrinkBasenameSize = 255 - \strlen($uniqExtension);
+            $shortBasename = \substr($originalBasename, 0, $shrinkBasenameSize);
+
+            return \sprintf('%s%s', $shortBasename, $uniqExtension);
+
+        // The extension is too long, but first try to preserve the basename, if possible
+        // 253 is used to account for a dot and one letter extension
+        } elseif (\strlen($uniqBasename) <= 253) {
```

Comment:
No need for an "else" after a return

Reply:
Right! Sorry about that, I'll have that changed in just a moment.

Reply:
This has been fixed in commit [3cecc4c](https://github.com/aewens/VichUploaderBundle/commit/3cecc4c3e4c0ef831872a1cf097fc6be1ed605fa#diff-43549d537b7d4f88517af8b366fd4a62)